Caricamento Norme di Progetto (#53)
Argo-swe/Docs: sources/documents/RTB/NormeDiProgetto/Supporto/documentazione.tex

\subsection{Documentazione}

\subsubsection{Descrizione}
Il processo di documentazione registra l'informazione generata da altri processi o attività. Il processo contiene l'insieme di attività che pianificano, producono, modificano, rilasciano e mantengono i documenti legati al progetto.\\
Il processo consiste nelle seguenti attività:
\begin{itemize}
  \item Implementazione del processo;
  \item Progettazione e sviluppo;
  \item Rilascio.
\end{itemize}

\paragraph{Implementazione del processo}\label{implementazioneprocessodocumentazione}
Questa attività definisce quali documenti saranno generati durante il progetto, definendo per ciascuno:
\begin{itemize}
  \item Titolo;
  \item Scopo;
  \item Descrizione;

>   \item Responsabilità per input, redazione, verifica e approvazione;

Reviewer: "input" in che senso?

Author: inteso come "fornire contributo", ad esempio nei casi in cui non è il solo redattore ad essere in possesso delle informazioni necessarie. Sostituirò con "contribuzione" per essere più chiaro

  \item Pianificazione per versioni provvisorie e finali.
\end{itemize}

\paragraph{Progettazione e sviluppo}

> Questa attività consiste nel progettare e redarre ciascun documento nel rispetto degli standard definiti per formato e contenuto, controllati dal \glossario{Verificatore}.

Reviewer: ```suggestion
Questa attività consiste nel progettare e redarre ciascun documento nel rispetto degli standard definiti per formato e contenuto, successivamente controllati dal \glossario{Verificatore}.
```

\paragraph{Rilascio}
Questa attività comincia con l'approvazione finale del documento da parte del \glossario{Responsabile} in carica, e della Proponente nel caso di verbali ad uso esterno. Prosegue con la pubblicazione del documento nel \glossario{repository} apposito della documentazione.

\subsubsection{Lista documenti}
I documenti da produrre e mantenere durante il corso del progetto sono:
\begin{itemize}
  \item \emph{Piano di Progetto};
  \item \emph{Norme di Progetto};
  \item \emph{Piano di Qualifica};
  \item \emph{Analisi dei Requisiti};
  \item \emph{Manuale Utente};
  \item \emph{Glossario};
  \item \emph{Verbali Interni};
  \item \emph{Verbali Esterni}.
\end{itemize}

\subsubsection{Ciclo di vita}
Il ciclo di vita di un documento è composto dai seguenti eventi:
\begin{enumerate}

>   \item Vengono definite le caratteristiche di base del documento o di una sua parte come da \ref{implementazioneprocessodocumentazione};

Reviewer: ```suggestion
  \item Vengono definite le caratteristiche di base del documento o di una sua parte come da sezione \ref{implementazioneprocessodocumentazione};
```

  \item Il Redattore stila una bozza iniziale. Se è necessario l'input di più persone in maniera sincrona, tale bozza viene prodotto in un ambiente condiviso;
  \item Prodotta una bozza di tutto il contenuto necessario, il Redattore produce una versione del documento con la forma e i metodi stabiliti in queste norme;
  \item Viene sottoposto a verifica il risultato della redazione. Se il Verificatore propone delle modifiche, vengono attuate ritornando alla fase precedente;

>   \item In seguito a un esito positivo della verifica, se il risultato è un documento completo e che richiede rilascio, viene sottoposto a una approvazione finale del responsabile, bloccante in modo analogo alla verifica.

Reviewer: ```suggestion
  \item In seguito a un esito positivo della verifica, se il risultato è un documento completo e che richiede rilascio, viene sottoposto ad un'approvazione finale del responsabile, bloccante in modo analogo alla verifica.
```

  \end{itemize}
  \item Riunione:
  \begin{itemize}
    \item Ordine del giorno dell'incontro;
    \item Discussione e decisioni prese durante l'incontro, contiene il corpo principale del verbale.
  \end{itemize}
  \item Tabella di task ToDo/In progress:
  \begin{itemize}
    \item Codice della \glossario{issue} \glossario{GitHub} relativa all'incarico;
    \item Incarico;
    \item Incaricato/a;
    \item Scandenza.
  \end{itemize}
\end{enumerate}

\subsubsection{Stile}
Di seguito sono elencate la convenzioni stilistiche adottate dalla documentazione del gruppo.

\paragraph{Utilizzo del femminile}
Quando è necessario fare riferimento tramite ruolo di progetto ad un membro del gruppo con il genere femminile, si utilizzano i seguenti termini:
\begin{itemize}
  \item \textbf{Responsabile} è invariato;
  \item \textbf{Amministratrice} al posto di Amministratore;
  \item \textbf{Analista} è invariato;
  \item \textbf{Progettista} è invariato;
  \item \textbf{Programmatrice} al posto di Programmatore;
  \item \textbf{Redattrice} al posto di Redattore;
  \item \textbf{Verificatrice} al posto di Verificatore.
\end{itemize}

\paragraph{Formattazione testo}
\begin{itemize}

>   \item \textbf{Termini nel Glossario:} Indicati alla prima occorrenza di un documento in \textit{corsivo} e con una \textit{\small{G}} a fine parola;

Reviewer: reminder: verificare se ancora corretto dopo discussione al ddb del 26/4

  \item \textbf{Nomi di documento:} Indicati in \textit{corsivo} con le iniziali di parola maiuscole eccetto preposizioni (es.: \textit{Piano di Progetto}, non \textit{Piano Di Progetto});
  \item \textbf{Nomi di ruolo:} Indicati con la iniziale maiuscola e in \textit{corsivo};

>   \item \textbf{Data:} Indicata in formato YYYY-MM-DD.

Reviewer: ```suggestion
  \item \textbf{Data:} Indicata in formato YYYY-MM-DD nelle tabelle riassuntive e nei nomi dei file, in formato esteso (esempio: 20 aprile 2024) quando si trova all'interno di testo discorsivo.
```
